### PR TITLE
NO-ISSUE: Add natsort package to requirements.txt

### PR DESCRIPTION
**Pull request by bkopilov**
The package allows us to sort complicated versions with numbers and naming. Using this package in custom ocp tests, getting a list of versions and sorting for last element .

Package info: https://pypi.org/project/natsort/

### CODE REVIEW

**Comment by bkopilov** [comment]
@adriengentil , @danmanor

**Comment by danmanor** [comment]
@bkopilov Why do you add the package but not use it ?

File: requirements.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ junit-report==0.2.7
 kubernetes==29.0.0
 libvirt-python==10.2.0
 munch==4.0.0
+natsort
```
Comment (danmanor) [inline requirements.txt:13]: We need the exact version

**Comment by bkopilov** [comment]
> @bkopilov Why do you add the package but not use it ?

I am using it in kni-assisted-installer-auto tests which run from podman containers.

**Comment by danmanor** [comment]
@bkopilov I don't understand, it is not used in this repo, so why add it here ?

**Comment by bkopilov** [comment]
> @bkopilov I don't understand, it is not used in this repo, so why add it here ?

Because here we controll what pip packages are installed on build image.
Dockerfile.assisted-test-infra

> RUN pip3 install --upgrade pip && \
>       pip3 install --no-cache-dir -I -r ./requirements.txt -r ./requirements-dev.txt && \
>       pip3 install --upgrade /build/pip/*

**Comment by danmanor** [comment]
I mean why not add it to `kni-assisted-installer-auto`  ?

**Comment by eliorerz** [comment]
> I mean why not add it to `kni-assisted-installer-auto` ?

I agree, if it's not used in test-infra I think the best it to add it as a dependency to `kni-assisted-installer-auto`

**Comment by bkopilov** [comment]
> > I mean why not add it to `kni-assisted-installer-auto` ?
> 
> I agree, if it's not used in test-infra I think the best it to add it as a dependency to `kni-assisted-installer-auto`

Hi , 
the kni-assisted-installer run from the container (Centos9) the package is part of pip installation on the container.
The Dockerfile.assisted-test-infra running the requirements.txt installation .

How can i install the python package from KNI ? 
Do you mean 
import pip 
pip.main(['install', "natsort"])

**Comment by bkopilov** [comment]
> > > I mean why not add it to `kni-assisted-installer-auto` ?
> > 
> > 
> > I agree, if it's not used in test-infra I think the best it to add it as a dependency to `kni-assisted-installer-auto`
> 
> Hi , the kni-assisted-installer run from the container (Centos9) the package is part of pip installation on the container. The Dockerfile.assisted-test-infra running the requirements.txt installation .
> 
> How can i install the python package from KNI ? Do you mean import pip pip.main(['install', "natsort"])

@eliorerz , @lalon4  , Not sure how can i force this package from kni tests , the image docker set by you before we run kni ... and i must update the container package where the code runs from ?

What do you suggest  ?  the make action is from test-infra...  should i call to pip from KNI ?

**Comment by eliorerz** [comment]
> > > > I mean why not add it to `kni-assisted-installer-auto` ?
> > > 
> > > 
> > > I agree, if it's not used in test-infra I think the best it to add it as a dependency to `kni-assisted-installer-auto`
> > 
> > 
> > Hi , the kni-assisted-installer run from the container (Centos9) the package is part of pip installation on the container. The Dockerfile.assisted-test-infra running the requirements.txt installation .
> > How can i install the python package from KNI ? Do you mean import pip pip.main(['install', "natsort"])
> 
> @eliorerz , @lalon4 , Not sure how can i force this package from kni tests , the image docker set by you before we run kni ... and i must update the container package where the code runs from ?
> 
> What do you suggest ? the make action is from test-infra... should i call to pip from KNI ?

I think that the best approach here is to add another step during kni-auto test-infra image build process, then build a `qe-test-infra` image using a new Dockerfile with all the dependencies you need.

**Comment by bkopilov** [comment]
@eliorerz , Ok . i will close it and move the logic to kni-test-infra

**Comment by bkopilov** [comment]
Moving the logic to kni-test

File: requirements.txt [hunk 2]
```diff
@@ -10,6 +10,8 @@ junit-report==0.2.7
 kubernetes==29.0.0
 libvirt-python==10.2.0
 munch==4.0.0
+# natsort used in kni-assisted-installer-auto
+natsort==8.4.0
```
Comment (eliorerz) [inline requirements.txt:14]:
```suggestion
natsort==8.4.0  # natsort used in kni-assisted-installer-auto
```

**Review by eliorerz — CHANGES_REQUESTED** [review]

**Comment by eliorerz** [comment]
/retitle NO-ISSUE: Add natsort package to requirements.txt

**Review by eliorerz — APPROVED** [review]
LGTM

**Pull request merged**